coap-gateway/coapconv/coapconv.go

```diff
@@ -322,7 +338,22 @@ func NewNotifyResourceChangedRequestsFromBatchResourceDiscovery(deviceID, connec
 			},
 			CommandMetadata: metadata,
 			Status:          code,
-		})
+			Etag:            r.ETag,
+		}
+		if len(etag) > 0 && bytes.Equal(etag, r.ETag) {
+			latestETagResource = resourceChangedReq
+			continue
+		}
+		requests = append(requests, resourceChangedReq)
+	}
+	// send latestETagResource need to be send as last because the resource are applied in order in resource aggregate
+	// so latestETagResource is the last resource in the batch
+	if latestETagResource != nil {
+		requests = append(requests, latestETagResource)
+	} else if len(requests) > 0 && len(etag) > 0 {
+		// if there is no latestETagResource we need to set etag to the last resource in the batch
+		// to make sure that the one resource contains the etag
+		requests[len(requests)-1].Etag = etag
```

Comment:
is it guaranteed that `len(requests)` is at least 1?

Reply:
What's the advantage of this instead of returning error if latestETAgResource is not found?

Reply:
`if len(requests) > 0`  guarantees that the size > 0
If someone don't use iotivity-lite and the etag is is not set per resource so we need to put the etag to some resource, otherwise etag will be not used at all for such device.